dali/pipeline/data/types.h

```diff
@@ -92,6 +92,17 @@ enum DALIDataType {
 // Dummy type to represent the invalid default state of dali types.
 struct NoType {};
 
+template <typename T>
+struct NormalizedType { using type = T; };
+
+// This way SetType<float16_cpu> will set SetType<float16>
```

Comment:
Maybe add info around here or `SetType`, that this will cause that both `float16_cpu` and `float16` will be handled internally (by python interface?) as the same type `float16`, as they have the same binary representation, and this still allows to access buffers by `float16_cpu` pointers.